### PR TITLE
feat(plasma-web): Add optional blur prop for Modal

### PR DESCRIPTION
**Pull request by Yakutoc**
Необходимо сделать **blur** подложки модального окна опциональным.

**По умолчанию** используем токен - `background-color: overlaySoft`

Было:
Снимок экрана 2023-02-16 в 13 09 27

Стало: 
Снимок экрана 2023-02-16 в 13 09 14
   

:baby_chick: Download canary assets:

[colors--canary.347.4220157206.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.347.4220157206.xml)  
[color_metro_ios-swift--canary.347.4220157206.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.347.4220157206.swift)  
[color_metro_kotlin--canary.347.4220157206.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.347.4220157206.kt)  
[color_metro_react-native--canary.347.4220157206.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.347.4220157206.ts)  
[color_sberHealth_ios-swift--canary.347.4220157206.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.347.4220157206.swift)  
[color_sberHealth_kotlin--canary.347.4220157206.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.347.4220157206.kt)  
[color_sberHealth_react-native--canary.347.4220157206.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.347.4220157206.ts)  
[color_sbermarket_ios-swift--canary.347.4220157206.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.347.4220157206.swift)  
[color_sbermarket_kotlin--canary.347.4220157206.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.347.4220157206.kt)  
[color_sbermarket_react-native--canary.347.4220157206.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.347.4220157206.ts)  
[color_sberprime_ios-swift--canary.347.4220157206.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.347.4220157206.swift)  
[color_sberprime_kotlin--canary.347.4220157206.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.347.4220157206.kt)  
[color_sberprime_react-native--canary.347.4220157206.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.347.4220157206.ts)  
[color_selgros_ios-swift--canary.347.4220157206.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.347.4220157206.swift)  
[color_selgros_kotlin--canary.347.4220157206.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.347.4220157206.kt)  
[color_selgros_react-native--canary.347.4220157206.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.347.4220157206.ts)  
[color_smbusiness_ios-swift--canary.347.4220157206.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.347.4220157206.swift)  
[color_smbusiness_kotlin--canary.347.4220157206.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.347.4220157206.kt)  
[color_smbusiness_react-native--canary.347.4220157206.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.347.4220157206.ts)  
[PlasmaTokensColor--canary.347.4220157206.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.347.4220157206.swift)  
[shadow_sbermarket_react-native--canary.347.4220157206.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.347.4220157206.ts)  
[typo_mage_ios-swift--canary.347.4220157206.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.347.4220157206.swift)  
[typo_mage_kotlin--canary.347.4220157206.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.347.4220157206.kt)  
[typo_mage_react-native--canary.347.4220157206.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.347.4220157206.ts)  
[typo_plasma_ios-swift--canary.347.4220157206.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.347.4220157206.swift)  
[typo_plasma_kotlin--canary.347.4220157206.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.347.4220157206.kt)  
[typo_plasma_react-native--canary.347.4220157206.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.347.4220157206.ts)  
[typo_ruler_ios-swift--canary.347.4220157206.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.347.4220157206.swift)  
[typo_ruler_kotlin--canary.347.4220157206.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.347.4220157206.kt)  
[typo_ruler_react-native--canary.347.4220157206.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.347.4220157206.ts)  
[typo_sage_ios-swift--canary.347.4220157206.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.347.4220157206.swift)  
[typo_sage_kotlin--canary.347.4220157206.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.347.4220157206.kt)  
[typo_sage_react-native--canary.347.4220157206.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.347.4220157206.ts)  
[typo_sbermarket_ios-swift--canary.347.4220157206.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.347.4220157206.swift)  
[typo_sbermarket_kotlin--canary.347.4220157206.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.347.4220157206.kt)  
[typo_sbermarket_react-native--canary.347.4220157206.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.347.4220157206.ts)  
[typo_soulmate_ios-swift--canary.347.4220157206.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.347.4220157206.swift)  
[typo_soulmate_kotlin--canary.347.4220157206.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.347.4220157206.kt)  
[typo_soulmate_react-native--canary.347.4220157206.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.347.4220157206.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.143.0-canary.347.4220157206.0
  npm install @salutejs/plasma-hope@0.3.0-canary.347.4220157206.0
  npm install @salutejs/plasma-web@1.169.0-canary.347.4220157206.0
  # or 
  yarn add @salutejs/plasma-b2c@1.143.0-canary.347.4220157206.0
  yarn add @salutejs/plasma-hope@0.3.0-canary.347.4220157206.0
  yarn add @salutejs/plasma-web@1.169.0-canary.347.4220157206.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-347/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-347/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-347/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-347/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-347/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-347/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-347/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-347/

File: packages/plasma-web/api/plasma-web.api.md [hunk 1]
```diff
@@ -121,6 +123,7 @@ import { MaxLinesProps } from '@salutejs/plasma-core';
 import { Modal } from '@salutejs/plasma-hope';
 import { ModalProps } from '@salutejs/plasma-hope';
 import { ModalsProvider } from '@salutejs/plasma-hope';
+import { ModalView } from '@salutejs/plasma-hope';
```
Comment (Yeti-or) [inline packages/plasma-web/api/plasma-web.api.md:126]: А зачем его Экспортировать ?

File: packages/plasma-web/api/plasma-web.api.md [hunk 2]
```diff
@@ -121,6 +123,7 @@ import { MaxLinesProps } from '@salutejs/plasma-core';
 import { Modal } from '@salutejs/plasma-hope';
 import { ModalProps } from '@salutejs/plasma-hope';
```
Comment (Yeti-or) [inline packages/plasma-web/api/plasma-web.api.md:124]: что-то я понял что modalProps не показательны, мы не видим что изменилось в это api =/

File: packages/plasma-hope/src/components/Modal/Modal.tsx [hunk 3]
```diff
@@ -12,7 +12,13 @@ export interface ModalProps extends ModalViewProps {
      * Отображение модального окна.
      */
     isOpen: boolean;
+
+    /**
+     * Нужно ли применять blur для подложки
+     */
+    withBlur?: boolean;
```
Comment (Yeti-or) [inline packages/plasma-hope/src/components/Modal/Modal.tsx:19]: почему with ?

у нас есть всякие disabled / animated / outlined / stretch / autoscroll

File: packages/plasma-hope/src/components/Modal/Modal.tsx [hunk 4]
```diff
@@ -69,16 +75,25 @@ const StyledWrapper = styled.div<HidingProps>`
...
 `;
-const StyledOverlay = styled.div<{ transparent?: boolean }>`
+
+const StyledOverlay = styled.div<{ transparent?: boolean; $withBlur?: boolean }>`
     position: absolute;
 
+    ${({ $withBlur }) => {
+        return css`
+            --background-color: ${$withBlur ? darkOverlayBlur : overlaySoft};
+            --backdrop-filter: ${$withBlur ? 'blur(1rem)' : 'none'};
```
Comment (Yeti-or) [inline packages/plasma-hope/src/components/Modal/Modal.tsx:85]: а зачем это в css переменной лежит ?

File: packages/plasma-hope/src/components/Modal/Modal.tsx [hunk 5]
```diff
@@ -99,15 +114,18 @@ const NoScroll = createGlobalStyle`
...
-    const controller = React.useContext(ModalsContext);
-    const [, forceRender] = React.useState(false);
 
-    React.useEffect(() => {
+    const wrapperRef = useRef<HTMLDivElement | null>(null);
+    const portalRef = useRef<HTMLElement | null>(null);
+
+    const controller = useContext(ModalsContext);
+
+    const [, forceRender] = useState(false);
```
Comment (Yeti-or) [inline packages/plasma-hope/src/components/Modal/Modal.tsx:126]: почему это понадобилось ??

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-347/

File: packages/plasma-web/api/plasma-web.api.md (hunk 1 above)
Comment (Yakutoc) [reply packages/plasma-web/api/plasma-web.api.md:126]: @Yeti-or В мастере ошибка - https://plasma.sberdevices.ru/web-storybook/?path=/story/controls-modal--default  

Чиниться через импорт.

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-347/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-347/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-347/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-347/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-347/temple-storybook/

File: packages/plasma-hope/src/components/Modal/Modal.tsx (hunk 3 above)
Comment (Yakutoc) [reply packages/plasma-hope/src/components/Modal/Modal.tsx:19]: @Yeti-or Посчитал так как у нас будет boolean то лучше через префикс - 

Обыкновенно использую is, has или with.

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

File: packages/plasma-hope/src/components/Modal/Modal.tsx (hunk 3 above)
Comment (Yeti-or) [reply packages/plasma-hope/src/components/Modal/Modal.tsx:19]: а где  у нас ещё `with` ?

File: packages/plasma-hope/src/components/Modal/Modal.tsx (hunk 5 above)
Comment (Yakutoc) [reply packages/plasma-hope/src/components/Modal/Modal.tsx:126]: @Yeti-or Так было до всех моих изменений. 

Есть комментарий от @neretin-trike 

```
          /**
         * Изменение стейта нужно для того, чтобы модальное окно
         * отобразилось после записи DOM элемента в portalRef.current
         */
        forceRender(true);
```

File: packages/plasma-web/api/plasma-web.api.md (hunk 1 above)
Comment (Yeti-or) [reply packages/plasma-web/api/plasma-web.api.md:126]: то есть у нас ModalView это API ?

File: packages/plasma-hope/src/components/Modal/Modal.tsx (hunk 4 above)
Comment (Yakutoc) [reply packages/plasma-hope/src/components/Modal/Modal.tsx:85]: @Yeti-or Что именно смущает? 

Или ты к тому, что darkOverlayBlur вообще не нужен?

File: packages/plasma-web/api/plasma-web.api.md (hunk 2 above)
Comment (Yeti-or) [reply packages/plasma-web/api/plasma-web.api.md:124]: https://github.com/salute-developers/plasma/pull/349

File: packages/plasma-web/api/plasma-web.api.md (hunk 1 above)
Comment (Yakutoc) [reply packages/plasma-web/api/plasma-web.api.md:126]: Получается, что да. 

Раньше то работало потому на уровне web это лежало в './'

File: packages/plasma-hope/src/components/Modal/Modal.tsx (hunk 3 above)
Comment (Yakutoc) [reply packages/plasma-hope/src/components/Modal/Modal.tsx:19]: @Yeti-or  Как пример

withOverlay, withTransition

File: packages/plasma-hope/src/components/Modal/Modal.tsx (hunk 4 above)
Comment (Yakutoc) [reply packages/plasma-hope/src/components/Modal/Modal.tsx:85]: Еще как довод, что потом идет такая логика

` background-color: ${({ transparent }) => (transparent ? 'transparent' : 'var(--background-color)')};` 


И что бы упростить понимание какое значение токена брать, я решил что css переменные подойдут.

**Comment by gibushnev** [comment]
Ок

**Review by gibushnev — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-347/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-347/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-347/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-347/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-347/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-347/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-347/temple-storybook/

**Review by Yeti-or — APPROVED** [review]

**Review by neretin-trike — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-347/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-347/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-347/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-347/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-347/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-347/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-347/temple-storybook/

**Pull request merged**